specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json

```diff
@@ -1098,6 +1098,232 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/getReservationDetailsReport": {
+      "post": {
+        "tags": [
+          "ReservedInstances"
+        ],
+        "operationId": "GetReservationDetailsReport_ByBillingAccountId",
+        "description": "Lists the reservations details for provided date range asynchronously based on enrollment id.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
+        },
+        "x-ms-examples": {
+          "ReservationDetails": {
+            "$ref": "./examples/GetReservationDetailsReportByBillingAccount.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/enrollmentIdParameter"
+          },
+          {
+            "$ref": "#/parameters/startDateParameter"
+          },
+          {
+            "$ref": "#/parameters/endDateParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
```

Comment:
these async POSTs should also contain a "200" response with the final output of the command that the user expects to see when it is done

Reply:
But the implementation will never return 200 even if the operation is ready with the response since we have an exclusive API which operates synchronously. If it is required just for the swagger documentation, we can. Please let me know.

Reply:
Generate one of the SDKs using autorest and look at the code.  An operation marked x-ms-long-running-operation knows how to poll the location header and will do that automatically until it completes then return the response.  You have linter errors due to the terminal response code (200) not being included here with the final response definition

Reply:
This is also why you don't need explicit paths to poll status.  It does it automatically

Reply:
removed status api from the swagger.